referenceframe/urdf.go

```diff
@@ -0,0 +1,258 @@
+package referenceframe
+
+import (
+	"encoding/xml"
+	"fmt"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/golang/geo/r3"
+	"github.com/pkg/errors"
+
+	spatial "go.viam.com/rdk/spatialmath"
+)
+
+// URDFConfig represents all supported fields in a Universal Robot Description Format (URDF) file.
+type URDFConfig struct {
+	XMLName xml.Name `xml:"robot"`
+	Name    string   `xml:"name,attr"`
+	Links   []struct{
+		XMLName   xml.Name `xml:"link"`
+		Name      string   `xml:"name,attr"`
+		Collision []struct{
+			XMLName xml.Name `xml:"collision"`
+			Name    string   `xml:"name,attr"`
+			Origin  struct{
+				XMLName xml.Name `xml:"origin"`
+				RPY     string   `xml:"rpy,attr"`  // "r p y" format
```

Comment:
Will poses always be RPY euler angles?

Also, are these degrees or radians? A comment on this struct should say.

Reply:
Always RPY, always Euler angles, always radians, always meters. I can add all that detail to struct comments.

Reply:
Yes please :)